include/net/net_pkt.h

```diff
@@ -622,13 +622,17 @@ static inline void net_pkt_set_ipv6_fragment_start(struct net_pkt *pkt,
 
 static inline uint16_t net_pkt_ipv6_fragment_offset(struct net_pkt *pkt)
 {
-	return pkt->ipv6_fragment_offset;
+	return pkt->ipv6_fragment_flags & 0xfff8;
+}
+static inline uint8_t net_pkt_ipv6_fragment_more(struct net_pkt *pkt)
```

Comment:
should this return bool rather than uint8_t?

Reply:
This returns the value of the "more" bit. But you are right that it is assigned to a bool variable, so returning a bool could make more sense.